sdk/samples/iot/templates/iothub/test-resources-post.ps1

```diff
@@ -36,33 +59,33 @@ Get-Content -Path device_ec_cert.pem, device_ec_key.pem | Set-Content -Path devi
 openssl x509 -noout -fingerprint -in device_ec_cert.pem | % {$_.replace(":", "")} | % {$_.replace("SHA1 Fingerprint=", "")} | Tee-Object -FilePath fingerprint.txt
 $fingerprint = Get-Content -Path .\fingerprint.txt
 
+Write-Host "Waiting for active IoT Hub"
+$hub_obj = waitForActiveHub
+
 $retryCount = 0
 do
 {
   $retryCount++
-  Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds."
+  Write-Host "Adding cert device to the allocated hub: attempt #$retryCount"
   Start-Sleep -Seconds $retryCount
 
-  # Get the hub as an object
-  $hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
-}
-while ($retryCount -lt 8 -and $hub_obj.Properties.State -ne "Active")
-
-if ($hub_obj.Properties.State -ne "Active")
-{
-  Write-Warning "AzIotHub instance is not yet active so later steps might fail as a result. If they fail often you can increase the retry count above."
+  # Pass fingerprint to IoTHub
+  Add-AzIotHubDevice `
+  -InputObject $hub_obj `
+  -DeviceId $deviceID `
+  -AuthMethod "x509_thumbprint" `
+  -PrimaryThumbprint $fingerprint `
+  -SecondaryThumbprint $fingerprint `
+  -ErrorAction Continue
+
+  if ($? -ne $true)
+  {
+    Write-Host "Adding cert device failed: trying again."
+  }
 }
+while ($retryCount -lt 3 -and $? -ne $true)
```

Comment:
Nit: it would be useful to have all these retry counts at the beginning of the file (as a config section) to easily update later.